File: python/day5/main.py
student_scores = [3, 7, 12, 19, 24, 31, 42, 56, 67, 89]
# highest_number = max(student_scores)
# print(highest_number)

# highest_score = 0
# for score in student_scores:
#     if score > highest_score:
#         highest_score = score
# print(score)


# number_sum = 0
# for i in range(1,101):
#     number_sum += i
# print(number_sum)


# Project 5: RANDOM PASSWORD GENERATOR

from random import choice, randint, shuffle
from string import ascii_letters

print("Welcome to the PyPassword Generator!")
letters_count = int(input("How many letters would you like in your password? "))
symbols_count = int(input("How many symbols would you like in your password? "))
numbers_count = int(input("How many numbers would you like in your password? "))
lst = []
random_letters = ""
for _ in range(letters_count):  # Generate 10 random letters
    random_letters =  choice(ascii_letters)
    lst.append(random_letters)

symbols = "!@#$%^&*()-_+=[]{}|;:,.<>?/"
random_symbols = ""
for _ in range(symbols_count):
    random_symbols = choice(symbols)
    lst.append(random_symbols)    
    
random_numbers = ""
for _ in range(numbers_count):
    random_numbers = str(randint(0, 9))
    lst.append(random_numbers)
    

print(lst)
shuffle(lst)
print(lst)

print(f"Your password is: {lst}")

File: go/day5/main.go
package main

import (
	"fmt"
	"math/rand"
	"strings"
	"time"
)

func main() {
	rand.Seed(time.Now().UnixNano())

	fmt.Println("Welcome to the GoPassword Generator!")

	// Get user input
	var lettersCount, symbolsCount, numbersCount int
	fmt.Print("How many letters would you like in your password? ")
	fmt.Scan(&lettersCount)
	fmt.Print("How many symbols would you like in your password? ")
	fmt.Scan(&symbolsCount)
	fmt.Print("How many numbers would you like in your password? ")
	fmt.Scan(&numbersCount)

	letters := "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
	symbols := "!@#$%^&*()-_+=[]{}|;:,.<>?/"
	numbers := "0123456789"

	var password []string
	for i := 0; i < lettersCount; i++ {
		randomLetter := string(letters[rand.Intn(len(letters))])
		password = append(password, randomLetter)
	}

	for i := 0; i < symbolsCount; i++ {
		randomSymbol := string(symbols[rand.Intn(len(symbols))])
		password = append(password, randomSymbol)
	}

	for i := 0; i < numbersCount; i++ {
		randomNumber := string(numbers[rand.Intn(len(numbers))])
		password = append(password, randomNumber)
	}

	rand.Shuffle(len(password), func(i, j int) {
		password[i], password[j] = password[j], password[i]
	})
	finalPassword := strings.Join(password, "")
	fmt.Printf("Your password is: %s\n", finalPassword)
}
